### PR TITLE
Feature/get features and bugfixes

### PR DESCRIPTION
**Pull request by halvfigur**
Don't initialize json ctor params with {} 
Use OutgoingEvent for replies to calls
Add Event::GetFeatures() function

### CODE REVIEW

File: src/sdk/cpp/lib/include/context.hpp [hunk 1]
```diff
@@ -282,13 +282,14 @@ class CallContext : public EventContext {
     void GatherAndReply(gather_and_reply_func_ptr func, timeout_func_ptr timeout_func, Args... args) {
         auto now_ms = GetEpochTimeMs();
         auto tokens = std::vector<CallToken>{args...};
-        auto prepared_reply = PreparedFunctionReply{talent_id_, feature_, subject_, channel_, call_, return_topic_, publisher_};
+        auto prepared_reply = PreparedFunctionReply{talent_id_, feature_, event_, return_topic_, publisher_};
```
Comment (JochenKienzle) [inline src/sdk/cpp/lib/include/context.hpp:285]: Event carries more than these 3 attributes. Will this lead to more overhead?

**Review by JochenKienzle — CHANGES_REQUESTED** [review]
One finding (see comment)

**Review by argerus — APPROVED** [review]

File: src/sdk/cpp/lib/include/context.hpp (hunk 1 above)
Comment (halvfigur) [reply src/sdk/cpp/lib/include/context.hpp:285]: There's small overhead but not providing the event as an argument will add an additional five arguments to the `PreparedFunctionReply` constructor. I'll change it if you insist but maybe it's better to leave it as is.

**Comment by halvfigur** [comment]
Closing because all the features in this PR are in [PR 94](https://github.com/GENIVI/iot-event-analytics/pull/94) which is being actively reviewed.

**Pull request closed without merging**